packages/react-data-grid/src/Canvas.js

```diff
@@ -153,41 +153,43 @@ const Canvas = React.createClass({
   getSubRows(row) {
     let subRowDetails = this.props.getSubRowDetails(row);
     if (subRowDetails.expanded === true) {
-      return subRowDetails.children.map(r => {
-        return { row: r };
+      return subRowDetails.children.map((r, i) => {
+        return { row: r, childIndex: i };
       });
     }
   },
 
-  addSubRows(rowsInput, row, i, displayEnd, treeDepth) {
+  addSubRows(rowsInput, row, i, displayEnd, treeDepth, siblingIndex, numberSiblings) {
     let subRowDetails = this.props.getSubRowDetails(row) || {};
     let rows = rowsInput;
     let increment = i;
     if (increment < displayEnd) {
       subRowDetails.treeDepth = treeDepth;
```

Comment:
instead of setting everything by hand wouldn't it be cleanear to use object assign and do all of this in 1 line of code?